bump ratatui to 0.30.0

fujiapple852 opened the pull request:
See https://github.com/ratatui/ratatui/releases/tag/ratatui-v0.30.0

fujiapple852 commented:
Blocked on MSRV:

```
Crate source was found to be incompatible with Rust version '1.78.0
```

orhun commented on crates/trippy-tui/src/frontend.rs:
```diff
@@ -69,7 +70,10 @@ enum ExitAction {
 }
 
 #[expect(clippy::too_many_lines, clippy::cognitive_complexity)]
-fn run_app<B: Backend>(terminal: &mut Terminal<B>, app: &mut TuiApp) -> io::Result<ExitAction> {
+fn run_app(
+    terminal: &mut Terminal<CrosstermBackend<Stdout>>,
+    app: &mut TuiApp,
+) -> io::Result<ExitAction> {
```
Looks correct to me... I don't remember the change led to this though. Can you share some references that requires this change?

j-g00da replied:
In 0.30 `io::Error` is no longer explicitly used for `Backend` errors, built-in backends still use `io::Error`, so this should in _most cases_ be backward compatible, but when using the `Backend` trait directly, the associated error type is unknown, so it requires additional bound (as you did).

j-g00da replied:
> You can probably use [`DefaultTerminal`](https://docs.rs/ratatui/latest/ratatui/type.DefaultTerminal.html)

Oh I didn't see your comment, yeah: https://github.com/fujiapple852/trippy/pull/1588#discussion_r2128219852

j-g00da commented:
> Blocked on MSRV:
> 
> ```
> Crate source was found to be incompatible with Rust version '1.78.0`
> ```
> 
> Ratatui `0.30.0` is de-facto `1.85` (2024 edition): [`Cargo.toml`](https://github.com/ratatui/ratatui/blob/671c2b4fd4ee277ba0dc36d152d66c40cfa1f030/Cargo.toml#L27) though there seems to be a discrepancy with the [`BREAKING-CHANGES.md`](https://github.com/ratatui/ratatui/blob/671c2b4fd4ee277ba0dc36d152d66c40cfa1f030/BREAKING-CHANGES.md?plain=1#L18) file.
> 
> Trippy `0.14.0` is intended to have an MSRV of 1.81.

This is an error, the MSRV is 1.85.0
https://github.com/ratatui/ratatui/pull/1896

fujiapple852 commented:
@j-g00da @joshka `DefaultTerminal` works and is much cleaner, thank you.

joshka commented:
This definitely needs rust 1.85 now due to the 2024 edition changes.

fujiapple852 commented:
@joshka yes, i'm considering reducing Trippy's Rust version policy from 12 months down to 6 months, which would put the next release on 1.85, roughly. 

It is a balance between trying to use an old enough toolchain to be supported on various distros and new enough to keep up with the Rust library ecosystem. I've found that in practice 12 months is to long, the Rust world moves on to quickly (especially around the edition bump) and so I think 6 months may be a better balance.

joshka commented:
Our philosophy in Ratatui is that we generally won't update the required version beyond N-2. This gives people at least 3 months to work out what they're doing and doesn't artificially prevent us from taking advantage of new things for too long (e.g. we waited until 1.87 was out to flip over to the 2024 edition). Obviously your use cases aren't ours.

> It is a balance between trying to use an old enough toolchain to be supported on various distros and new enough to keep up with the Rust library ecosystem. I've found that in practice 12 months is to long, the Rust world moves on to quickly (especially around the edition bump) and so I think 6 months may be a better balance.

Good news - Debian Trixie looks like it will support 1.85. but by the time it comes out I'd guess Ratatui will likely have moved past that version...

That said, I'm very apprehensive about distros pinning to versions like this. Artificial limitations on calling functions that have been available for years (rust 1.63 was released in 2022), are a game that I choose not to participate in unless forced to. 

Case in point, I had a job interview the other day where one part of the solution was a simple method introduced in the std lib in 1.80, but coderpad runs 1.72, so I had to polyfill the impl live. That was somewhat annoying.

fujiapple852 commented:
Updated to `0.30.0-beta.0`.

Note that Trippy is now on Rust 1.85 (2024) edition.

dependabot[bot] committed:
chore: bump `ratatui` to `0.30.0`

Bumps [tun](https://github.com/meh/rust-tun) from 0.8.4 to 0.8.5.
- [Release notes](https://github.com/meh/rust-tun/releases)
- [Commits](https://github.com/meh/rust-tun/commits/v0.8.5)

---
updated-dependencies:
- dependency-name: tun
  dependency-version: 0.8.5
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
